MIKKI-28

swjoon opened the pull request:
<!--
  템플릿은 아직 PR 작성이 익숙하지 않으신 분들을 위해서 제공하는 가이드입니다!
  리뷰어 또는 이 PR을 보게 될 다른 사람들이 이 PR을 보는데 참고할 수 있는 내용이 있다면 포함해서 작성해주시면 됩니다.
-->

## 📌 과제 설명 <!-- 어떤 걸 만들었는지 대략적으로 설명해주세요 -->
- 선상 낚시 게시글 API
  - Create
## 👩‍💻 요구 사항과 구현 내용 <!-- 기능을 Commit 별로 잘개 쪼개고, Commit 별로 설명해주세요 -->
- 선상 낚시 게시글 도메인 기본 구조 작성
- 선상 낚시 게시글 작성 기능
- 선상 낚시 게시글 작성 기능 테스트 코드 작성
  - ServiceTest
  - RepositoryTest

swjoon committed:
Feat: 선상 낚시 도메인 exception 적용 (MIKKI-28)

swjoon committed:
Feat: 선상낚시 게시글 create dto 생성 (MIKKI-28)

swjoon committed:
Feat: 선상낚시 게시글 entity 수정 및 from 메소드 추가 (MIKKI-28)

swjoon committed:
Feat: 선상낚시 게시글 repository 생성 & save 메소드 추가 (MIKKI-28)

swjoon committed:
Feat: 선상낚시 게시글 service 생성 & save 메소드 추가 (MIKKI-28)

swjoon committed:
Feat: 선상낚시 게시글 응답 메세지 상수 관리 (MIKKI-28)

swjoon committed:
Feat: 선상낚시 게시글 controller 생성 & 게시글 create API 추가 (MIKKI-28)

swjoon committed:
Fix: QueryDsl config 에 public 추가 (MIKKI-28)

swjoon committed:
Test: 선상낚시 게시글 create test code 작성 (MIKKI-28)
- ServiceTest
- RepositoryTest
- Util

Janghyeonsuk commented on backend/src/main/java/com/backend/domain/shipfishposts/repository/ShipFishPostsRepository.java:
```diff
@@ -0,0 +1,5 @@
+package com.backend.domain.shipfishposts.repository;
+
+public interface ShipFishPostsRepository {
+
```
```java
ShipFishPosts save(ShipFishPosts shipFishPosts);
```
메서드가 들어가야할거같습니다!

swjoon committed:
fix: repository & service 메소드 일부 수정 (MIKKI-28)

swjoon committed:
Test: 소스코드 수정 사항 테스트 코드 적용 (MIKKI-28)
- repository & service 일부 수정

kangjjs commented on backend/src/main/java/com/backend/domain/shipfishposts/controller/ShipFishPostsController.java:
```diff
@@ -0,0 +1,42 @@
+	private final ShipFishPostsService shipFishPostsService;
+
+	@PostMapping
+	public GenericResponse<?> createShipFishPost(
```
저희가 api명세서를 작성하긴했지만, swagger도 사용하기로해서 swagger를 통해서 문서화도 하면 좋을거같습니다.

kdo1999 replied:
타입 명시 Void로 수정 부탁드립니당

kdo1999 commented on backend/src/main/java/com/backend/domain/shipfishposts/controller/ShipFishPostsController.java:
```diff
@@ -0,0 +1,42 @@
+
+		if (bindingResult.hasErrors()) {
+			throw new GlobalException(GlobalErrorCode.NOT_VALID);
+		}
```
제외 하겠습니다.

kdo1999 commented on backend/src/main/java/com/backend/domain/shipfishposts/controller/ShipFishPostsController.java:
```diff
@@ -0,0 +1,42 @@
+
+		Long shipFishPostsId = shipFishPostsService.save(requestDto);
+
+		return GenericResponse.ok(ShipFishPostsMessageConstant.CREATE_POSTS_SUCCESS);
```
데이터 빼고 보내주세요~

kdo1999 replied:
ResponseEntity로 반환 부탁드립니당

kdo1999 commented on backend/src/main/java/com/backend/domain/shipfishposts/dto/request/ShipFishPostsRequest.java:
```diff
@@ -0,0 +1,43 @@
+
+public class ShipFishPostsRequest {
+
+	/**
```
위에 json으로 작성하고 아래 param 옆에는 설명만 달아주시면 됩니당

kdo1999 commented on backend/src/main/java/com/backend/domain/shipfishposts/repository/ShipFishPostsRepositoryImpl.java:
```diff
@@ -0,0 +1,21 @@
+
+	@Override
+	public ShipFishPosts save(ShipFishPosts shipFishPosts) {
+		return shipFishPostsJpaRepository.save(shipFishPosts);
```
띄워주세요

swjoon committed:
Fix: ShipFishPost exception 파일들 줄바꿈 (MIKKI-28)

swjoon committed:
Fix: ShipFishPost Repository 파일들 줄바꿈 및 주석 추가 (MIKKI-28)

swjoon committed:
Fix: ShipFishPost Data 수정 (MIKKI-28)
- entity from 메서드 converter 적용
- dto swagger @Schema 적용 및 @vaild 메세지 추가
- converter 클래스 생성

swjoon committed:
Fix: ShipFishPost Service save 메서드 converter 적용 (MIKKI-28)

swjoon committed:
Fix: ShipFishPost Controller swagger 적용 (MIKKI-28)

swjoon committed:
Test: ShipFishPost test code fixture-monkey 적용 (MIKKI-28)

kangjjs commented on backend/src/main/java/com/backend/domain/shipfishposts/controller/ShipFishPostsController.java:
```diff
@@ -0,0 +1,43 @@
+
+	private final ShipFishPostsService shipFishPostsService;
+
+	@Tag(name = "선상낚시 게시글 > 게시글 생성 API 명세")
```
저희 예시는 제 프로젝트는 PA 5.1 처럼 그 번호대로 나눠져서 컨트롤러 내에서도 서로 다른 번호를 가지고있던거라 Tag를 나눈거라 저희는 그냥 Controller별로 @Tag 관리하면 될거같습니다

swjoon committed:
Fix: ShipFishPost Controller @Tag 위치 수정 (MIKKI-28)

kdo1999 commented:
Github 연동 테스트

kdo1999 commented:
Github 연동 테스트 2

swjoon committed:
Fix: ShipFishPost Data 일부 수정 (MIKKI-28)
- ShipFishPosts
- ShipFishPostsConverter
- ShipFishPostsRequest

swjoon committed:
Fix: ShipFishPost Service 일부 수정 (MIKKI-28)
- duration 메서드 분리

swjoon committed:
Test: ShipFishPost Test 일부 수정 (MIKKI-28)
- BaseTest 파일 위치 변경
- test data 수정

swjoon committed:
fix: GlobalControllerAdvice 정리 (MIKKI-28)

swjoon committed:
feat: ShipFishPosts 조회수 컬럼 추가 (MIKKI-28)

swjoon committed:
fix: ShipFishPosts -> ShipFishingPost 로 rename 및 추가 (MIKKI-28)

swjoon committed:
fix: ShipFishingPost controller 수정 (MIKKI-28)
